monash-human-power/raspicam
# Display error message on overlay when DAS is disconnected

## Description

WOW ok I finished this PR quicker than I expected. GReat lol

When this PR is merged, it should:
- Display error message that the DAS is not connected to the overlay, or the broker has not initiated. 
  - Error message should disappear **only** when DAS is connected.
  - If DAS disconnects during the 'race', then the error message will reappear. 
  - Reconnect delay should also be changed to every 10 seconds. 
- Display data titles, even though data is not being sent to the overlay, when it didn't before.
- Add comment headers for the `Overlay` constructor in hopes that this will be more readable and organised.

The following changes are only applied to `overlay_new.py` at the moment. But I'm open to changing these for the other overlays if needed. 

ALSO, I'm open to any name changes for the new class `DASDisconnectMessage`, because it's quite long. But if everyone is cool with this, then I'm cool with it. 

## Screenshots

Initial version:
![image](https://user-images.githubusercontent.com/58575113/93505262-fb343980-f95d-11ea-840e-724ef2e8a728.png)

Updated version:
![image](https://user-images.githubusercontent.com/58575113/93604830-45262980-fa09-11ea-80f2-73d65b63591e.png)

## Steps to Test
Ensure that a `mosquitto` broker is **not** running in a terminal nor in the background. 

1. Run `poetry install`.
2. Run `poetry shell`.
3. Run `python overlay_new.py`.

To test if error message will disappear:

4. Run `mosquitto` in a terminal. Error message should disappear as soon as overlay is connected.
